ansible/roles/consul-haproxy/tasks/main.yml

```diff
@@ -4,10 +4,20 @@
     mode: 0644
     src: "haproxy.json.j2"
     dest: "/etc/consul.d/haproxy.json"
+  register: haproxy_json
 
-- name: Enable consul service # noqa ignore-errors
+- name: Make sure consul service is running
   ansible.builtin.systemd:
     name: consul
     state: started
     enabled: true
   ignore_errors: true
+  when: not haproxy_json.changed
+
+- name: Restart consul if service file changed
```

Comment:
This is best done as a handler, with a notify from the previous step

Reply:
fixed